File: .guides/tests/py-14.js

var fileName = window.location.origin + '/public/py/' + window.testEnv.cmd + '.py';

$.get(fileName)
.success(function(data) {

  if(data == 404) {
    console.log(exception);
    codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.INVALID, fileName + ' not found'); 
  }

  try {        
    var module = Sk.importMainWithBody("", false, data);

    if(message.trim() == 'Mary had a little lamb') {
      codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.SUCCESS, 'Well done!');
    }
    else {
      codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.FAILURE, 'Not quite right, try again!');
    }
  }
  catch(exception) {
    console.log('skulpt exception');
    console.log(exception);
    codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.INVALID, exception.toString()); 
  }
})
.fail(function(jqxhr, settings, exception) {
  console.log('jqxhr fail');
  console.log(exception);
  codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.INVALID, exception.message); 
});    

File: .guides/tests/py-14.py

maze = False
message = ''

def createEmptyMaze():
  global maze
  maze = True

def showMessage(m):
  global message
  message = m
  
try:
  execfile('/home/codio/workspace/public/py/py-14.py')
    
  if maze == True and message == 'Mary had a little lamb':
    print 'well done'
    exit(0)
except (IOError, SyntaxError) as e:
  pass
  
print 'Not quite right, try again!'
exit(1)
